Hazel/Dtls/Handshake.cs

```diff
@@ -405,6 +405,67 @@ public void Encode(ByteSpan span)
         }
     }
 
+    /// <summary>
+    /// Encode/Decode session information in ClientHello
+    /// </summary>
+    public struct HazelDtlsSessionInfo
+    {
+        public const byte CurrentClientSessionSize = 1;
+        public const byte CurrentClientSessionVersion = 1;
+
+        public byte FullSize => (byte)(1 + this.PayloadSize);
+        public byte PayloadSize;
+        public byte Version;
+
+        public HazelDtlsSessionInfo(byte version)
+        {
+            this.Version = version;
+            switch (version)
+            {
+                case 0: // Does not write version byte
+                    this.PayloadSize = 0;
+                    return;
+                case 1: // Writes version byte only
+                    this.PayloadSize = 1;
+                    return;
+            }
+
+            throw new ArgumentOutOfRangeException("Unimplemented Hazel session version");
+        }
+
+        public void Encode(ByteSpan writer)
+        {
+            writer[0] = this.PayloadSize;
+
+            if (this.Version > 0)
+            {
+                writer[1] = this.Version;
+            }
+        }
+
+        public static bool Parse(out HazelDtlsSessionInfo result, ByteSpan reader)
```

Comment:
Minor, but would recommend to change name to `TryParse` to match the standard signature pattern that is used here